presto-orc/src/main/java/com/facebook/presto/orc/reader/DoubleSelectiveStreamReader.java

```diff
@@ -247,6 +292,60 @@ private int readNoFilter(int[] positions, int positionCount)
     private int readWithFilter(int[] positions, int positionCount)
             throws IOException
     {
+        boolean testNull = (nonDeterministicFilter && this.filter.testNull()) || nullsAllowed;
+
+        if (useBatchMode()) {
+            int totalPositionCount = positions[positionCount - 1] + 1;
+            int readCount = 0;
+
+            final int filteredPositionCount;
+
+            if (presentStream == null) {
+                dataStream.next(values, totalPositionCount);
+
+                filteredPositionCount = evaluateFilter(positions, positionCount);
+
+                if (outputRequired && totalPositionCount > filteredPositionCount) {
+                    packLongs(values, outputPositions, filteredPositionCount);
+                }
+            }
+            else {
+                int nullCount = presentStream.getUnsetBits(totalPositionCount, nulls);
+
+                if (nullCount == totalPositionCount) {
+                    // all nulls
+                    if (testNull) {
+                        allNulls = true;
+                        filteredPositionCount = positionCount; // No positions were filtered out
+                    }
+                    else {
+                        filteredPositionCount = 0;
+                    }
+                }
+                else {
+                    // some nulls
+                    readCount = totalPositionCount - nullCount;
+                    dataStream.next(values, readCount);
+
+                    if (nullCount != 0) {
+                        // Note it should be totalPositionCount instead of positionCound
```

Comment:
typo in positionCound, would also be nice to tell why it should be the totalPositionCount